### ISSUE TITLE
Unable to open xDrip4iOS 4.4.15 (4121) and newer versions on iPad 2 (iOS 12.5.1)

### ISSUE THREAD

**Issue by parapenT1sta** [JohanDegraeve/xdripswift#175]
**Latest working version:** 4.4.14 (4120)
**Not working:** 4.4.15 / 4.5.0 / 4.5.1

**Issue:** xDrip4iOS closes automatically after I open it.

**Comment by JohanDegraeve** [comment]
is this in xCode ? or are you using the version from TestFlight ? in which case you have probably submitted a crash report ?

**Comment by parapenT1sta** [comment]
I'm using the TestFlight version, just sent the email through TestFlight.

**Comment by Cheun1** [comment]
i have also this problem 
**Latest working version**: 4.4.14 (4120)

**Comment by JohanDegraeve** [comment]
does someone here (who has the issue with iPad) have a mac ?
Can you do the following

- Install on your Mac : "Apple 2 configurator", it's in the App store. You don't need to launch it, only install
- Open a Terminal session  : click search (top left), type "Terminal" and double click
- type (copy) the following in the terminal session and press enter
/Applications/Apple\ Configurator\ 2.app/Contents/MacOS/cfgutil syslog | grep -i xDrip > ~/Downloads/result.txt

- connect your iPad to the Mac 
- If this would be the first time, you will probably need to allow your mac to access the ipad
- open xDrip4iOS (it will not launch)
- send result.txt to xdrip@proximus.be, it's in your Downloads folder

**Comment by parapenT1sta** [comment]
Do you think a virtual MAC machine will work? If yes, I'm willing to create one just to test it.

**Comment by JohanDegraeve** [comment]
I think so yes, if you can access the USB port

**Comment by parapenT1sta** [comment]
@JohanDegraeve I am installing it but btw, the latest version doesn't fix the issue with iPad.

**Comment by JohanDegraeve** [comment]
> @JohanDegraeve I am installing it but btw, the latest version doesn't fix the issue with iPad.

I know, btw it's not only iPad, it's also iPhone on iOS 12.x (that's what someone else told me)
But I don't have a device with iOS12.x, also on a simulator with iOS 12.x it works

so it would be useful if you can give me the info as described above on any iOS 12.x device

**Comment by Cheun1** [comment]
good news, I will have the TXT file tonight i keep you informed

**Comment by Cheun1** [comment]
sent

**Comment by bobhavlin** [comment]
Is the problem the missing NFC?

`Termination Description: DYLD, Library not loaded: /System/Library/Frameworks/CoreNFC.framework/CoreNFC  | Referenced from: /var/containers/Bundle/Application/12EA3371-4050-4EA7-AEC4-B692EB660E85/xdrip.app/xdrip | Reason: image not found
`

**Comment by JohanDegraeve** [comment]
that looks interesting info indeed, question now is how to solve it

**Comment by bobhavlin** [comment]
Maybe solution from [here](https://developer.apple.com/forums/thread/106057)?

"For iOS 12 you need to mark the CoreNFC framework as optional. That allows to run the app without crash"

**Comment by JohanDegraeve** [comment]
it's not even added

![image](https://user-images.githubusercontent.com/13840461/107096540-4961b080-680b-11eb-84c5-72b068f08842.png)

I found also this https://github.com/Alamofire/Alamofire/issues/3051

_"Commenting out use_frameworks and adding use_modular_headers! to the Pod file"_

**Comment by bobhavlin** [comment]
Maybe this helps:  https://stackoverflow.com/questions/44946057/dyld-library-not-loaded-rpath-corenfc-framework-corenfc-ios11-and-xcode-9-be

**Comment by JohanDegraeve** [comment]
I tried that, 4.5.12 uses that, doesn't solve it
https://github.com/JohanDegraeve/xdripswift/commits/fix175

**Comment by parapenT1sta** [comment]
@JohanDegraeve How to access to 4.5.12 version?

**Comment by JohanDegraeve** [comment]
@bobhavlin  could you take the log again with the version 4.5.12 ? (branch named fix175)
it seems to behave differently already

**Comment by bobhavlin** [comment]
Yes, fix175 runs on the iphone6s (ios 12.1.1) unlike the testflight version which crashed immediately.   Getting an error message, but probably due to my lack of developer account...

**Comment by JohanDegraeve** [comment]
what error message do you get ?

**Comment by bobhavlin** [comment]
To get it to build without a paid developer account, I have to delete the CoreNFC from the build...
![2021-02-07 13_05_53-macOS 11 0 - VMware Workstation 16 Player (Non-commercial use only)](https://user-images.githubusercontent.com/1026345/107156661-4965cb80-6945-11eb-8460-80402ec121c6.png)

**Comment by JohanDegraeve** [comment]
does it run if you delete CoreNFC ?

**Comment by bobhavlin** [comment]
I think I figured out how to get this to work and fully load (using branch named fix175)!

Using Xcode (12.2): 
- xdrip (target) -> Build Phases -> Link Binary With Libraries, I added CoreNFC.framework set to Optional
- xdrip (target) -> General, set CoreNFC.framework to "Do Not Embed"
- And the same two settings in Target xDrip4iOS Widget

It ran on my iphone6s (ios 12.1.1) without error.  

FYI, the Testflight version 4.5.14 (4136) crashed as in the original post.

[Source ](https://stackoverflow.com/questions/60655532/custom-framework-crashes-in-iphone-6-devices-due-to-nfc-feature-even-though-core)for the change
